tests/js/searchtools.js

```diff
@@ -94,6 +109,25 @@ describe('Basic html theme search', function() {
 
   });
 
+  describe('search result ranking', function() {
+
+    it('should score an object-name match above a page-title match', function() {
```

Comment:
... and a page-title match above above a subtitle match? Maybe easier to just say "it should rank things in the appropriate order" and then explain the details in a comment (maybe with a reference to #12391 for context)

Reply:
If possible I'd like to move in a more granular direction with these tests; breaking this into two:

  * Main document title should rank above section title (for equivalent/similar string matches).
  * Programming language domain object should rank above main document title (again for equivalent/similar string matches).

The problem with larger lists of "these things should rank in this order" is that it could lead to difficult-to-comprehend logic in the code that is more of an attempt by developers to keep the tests passing than to retain a set of sensible rules.  It does get tricky over time with either approach, but I think it's useful to be able to identify what specific rankings are broken (and when).

Links to GitHub issues and external resources can be a useful resource sometimes (like the duplicate-result test fixture data / TODO note recently) -- if we can though, I'd prefer to keep the code and tests self-contained and self-documenting.

Reply:
Okay! To be honest I don't have super strong opinions here as long as we have a test of some kind.

Reply:
> ... and a page-title match above above a subtitle match? [ ... snip ... ]

That's now in place more clearly, by splitting it out into a separate test case (and indirectly that also helped to find bugs in the `checkRanking` test utility function..).